src/ast.zig

```diff
@@ -54,6 +55,32 @@ fn fullPtrTypeComponents(tree: Ast, info: full.PtrType.Components) full.PtrType
     return result;
 }
 
+fn findMatchingRBrace(token_tags: []const TokenTag, l_brace: Ast.TokenIndex) Ast.TokenIndex {
+    std.debug.assert(token_tags[l_brace] == TokenTag.l_brace);
+
+    const start = l_brace + 1;
+    var depth: i32 = 0;
```

Comment:
Is it necessary to allow negative depth? Would it make sense to break early if there are too many closing braces?

Reply:
Yes, a negative depth would be a syntactic error and we could end on that. I can make that change.

Reply:
Oh, nevermind, I looked at the code and there is no way for there to be a negative depth. Depth starts at 0 and is only ever decremented when we see an r_brace, but not before checking if it is 0 and breaking if it is.

Reply:
Shouldn't the type be changed to `u32` or `usize` then?

Reply:
I decided to leave it as `i32` as a bit of protection against any future changes to the code that might make some use of negative values, in which case an unsigned integer type left as is could lead to a negative wrap around bug. That seemed to outweigh the extra precision of going to `u32`, since more than a billion tokens in a file unlikely, and even then I'd probably opt for `isize`.

I know zig has some protection against unsigned wrap around in some build modes, but coming from a C++ background I fear bugs related to unsigned wrapping, and I generally favor signed types if the integer is used for any arithmetic more complicated than simple incrementing.

That is my reasoning. I'll change it if you think it should still be changed.

Reply:
In contrast to C++, signed/unsigned integer overflow is safety-checked undefined behavior which means that you can safely choose the integer type that most closely represents your use-case and don't have to worry about UB or overflow bugs.
That being said, I'm starting to bikeshed so how about I approve this PR and we can go on with our lives. :)